### PR TITLE
Update Mavenplugin dependencies

### PR DESCRIPTION
**Pull request by MikeSchumacherCapgemini**
Updating core dependency in maven and maven test to fit the latest core release
Updating test dependencies to the latest releases of cobigen plugins

### CODE REVIEW

File: cobigen-maven/cobigen-maven-systemtest/src/test/resources/testdata/systemtest/Devon4JTemplateTest/TestClassLoadingWithTemplateFolder/pom.xml [hunk 1]
```diff
@@ -37,17 +37,17 @@
...
             <groupId>com.devonfw.cobigen</groupId>
             <artifactId>javaplugin</artifactId>
-            <version>7.0.0-SNAPSHOT</version>
+            <version>7.1.0-SNAPSHOT</version>
           </dependency>
           <dependency>
             <groupId>com.devonfw.cobigen</groupId>
             <artifactId>tsplugin</artifactId>
-            <version>7.0.0-SNAPSHOT</version>
+            <version>7.1.0-SNAPSHOT</version>
```
Comment (maybeec) [inline cobigen-maven/cobigen-maven-systemtest/src/test/resources/testdata/systemtest/Devon4JTemplateTest/TestClassLoadingWithTemplateFolder/pom.xml:50]: haven't we released the 7.1.0 now?

File: cobigen-maven/cobigen-maven-systemtest/src/test/resources/testdata/systemtest/Devon4JTemplateTest/TestClassLoadingWithTemplateFolder/pom.xml [hunk 2]
```diff
@@ -37,17 +37,17 @@
...
             <groupId>com.devonfw.cobigen</groupId>
             <artifactId>tempeng-freemarker</artifactId>
-            <version>7.0.0-SNAPSHOT</version>
+            <version>7.0.0</version>
           </dependency>
           <dependency>
             <groupId>com.devonfw.cobigen</groupId>
             <artifactId>javaplugin</artifactId>
-            <version>7.0.0-SNAPSHOT</version>
+            <version>7.1.0-SNAPSHOT</version>
```
Comment (maybeec) [inline cobigen-maven/cobigen-maven-systemtest/src/test/resources/testdata/systemtest/Devon4JTemplateTest/TestClassLoadingWithTemplateFolder/pom.xml:45]: haven't we released the 7.1.0 now?

File: cobigen-maven/cobigen-maven-systemtest/src/test/resources/testdata/systemtest/Devon4JTemplateTest/TestDifferentFileSystems/pom.xml [hunk 3]
```diff
@@ -37,47 +37,47 @@
...
             <groupId>com.devonfw.cobigen</groupId>
             <artifactId>openapiplugin</artifactId>
-            <version>7.0.0-SNAPSHOT</version>
+            <version>7.1.0-SNAPSHOT</version>
           </dependency>
           <dependency>
             <groupId>com.devonfw.cobigen</groupId>
             <artifactId>javaplugin</artifactId>
-            <version>7.0.0-SNAPSHOT</version>
+            <version>7.1.0-SNAPSHOT</version>
```
Comment (maybeec) [inline cobigen-maven/cobigen-maven-systemtest/src/test/resources/testdata/systemtest/Devon4JTemplateTest/TestDifferentFileSystems/pom.xml:55]: haven't we released the 7.1.0 now?

File: cobigen-maven/cobigen-maven-systemtest/src/test/resources/testdata/systemtest/Devon4JTemplateTest/TestDifferentFileSystems/pom.xml [hunk 4]
```diff
@@ -37,47 +37,47 @@
...
             <groupId>com.devonfw.cobigen</groupId>
             <artifactId>htmlplugin</artifactId>
-            <version>7.0.0-SNAPSHOT</version>
+            <version>7.0.0</version>
           </dependency>
           <dependency>
             <groupId>com.devonfw.cobigen</groupId>
             <artifactId>tsplugin</artifactId>
-            <version>7.0.0-SNAPSHOT</version>
+            <version>7.1.0-SNAPSHOT</version>
```
Comment (maybeec) [inline cobigen-maven/cobigen-maven-systemtest/src/test/resources/testdata/systemtest/Devon4JTemplateTest/TestDifferentFileSystems/pom.xml:70]: haven't we released the 7.1.0 now?

File: cobigen-maven/cobigen-maven-systemtest/src/test/resources/testdata/systemtest/Devon4JTemplateTest/TestDifferentFileSystems/pom.xml [hunk 5]
```diff
@@ -37,47 +37,47 @@
...
             <groupId>com.devonfw.cobigen</groupId>
             <artifactId>jsonplugin</artifactId>
-            <version>7.0.0-SNAPSHOT</version>
+            <version>7.0.0</version>
           </dependency>
           <dependency>
             <groupId>com.devonfw.cobigen</groupId>
             <artifactId>textmerger</artifactId>
-            <version>7.0.0-SNAPSHOT</version>
+            <version>7.1.0-SNAPSHOT</version>
```
Comment (maybeec) [inline cobigen-maven/cobigen-maven-systemtest/src/test/resources/testdata/systemtest/Devon4JTemplateTest/TestDifferentFileSystems/pom.xml:80]: haven't we released the 7.1.0 now?

File: cobigen-maven/cobigen-maven-systemtest/src/test/resources/testdata/systemtest/Devon4JTemplateTest/TestEntityInputDataaccessGeneration/pom.xml [hunk 6]
```diff
@@ -37,57 +37,57 @@
...
             <groupId>com.devonfw.cobigen</groupId>
             <artifactId>templates-devon4j</artifactId>
-            <version>3.1.8</version>
+            <version>20.08.002</version>
         </dependency>
         <dependency>
           <groupId>com.devonfw.cobigen</groupId>
           <artifactId>javaplugin</artifactId>
-          <version>7.0.0-SNAPSHOT</version>
+          <version>7.1.0-SNAPSHOT</version>
```
Comment (maybeec) [inline cobigen-maven/cobigen-maven-systemtest/src/test/resources/testdata/systemtest/Devon4JTemplateTest/TestEntityInputDataaccessGeneration/pom.xml:45]: haven't we released the 7.1.0 now?

Comment (maybeec) [reply cobigen-maven/cobigen-maven-systemtest/src/test/resources/testdata/systemtest/Devon4JTemplateTest/TestEntityInputDataaccessGeneration/pom.xml:45]: please check all dependencies. snapshots are always not stable and tend to fail after the next release when the snapshots are cleaned up from the nexus. If possible make all dependency to stable versions

**Review by maybeec — CHANGES_REQUESTED** [review]

File: cobigen-maven/cobigen-maven-systemtest/src/test/resources/testdata/systemtest/Devon4JTemplateTest/TestEntityInputDataaccessGeneration/pom.xml [hunk 7]
```diff
@@ -37,57 +37,57 @@
         <dependency>
             <groupId>com.devonfw.cobigen</groupId>
             <artifactId>templates-devon4j</artifactId>
-            <version>3.1.8</version>
```
Comment (maybeec) [inline cobigen-maven/cobigen-maven-systemtest/src/test/resources/testdata/systemtest/Devon4JTemplateTest/TestEntityInputDataaccessGeneration/pom.xml:40]: íf possible keep an old version of templates as well to also test backward compatibility

**Review by maybeec — CHANGES_REQUESTED** [review]
The target of this should not be to increase all versions.
The target should be to get tests running which are fail and just update versions which are necessary. You are adapting the maven tests, which should test the maven plugin and the maven plugin should not just run with the latest versions hopefully.

File: cobigen-maven/cobigen-maven-systemtest/src/test/resources/testdata/systemtest/Devon4JTemplateTest/TestEntityInputDataaccessGeneration/pom.xml (hunk 7 above)
Comment (MikeSchumacherCapgemini) [reply cobigen-maven/cobigen-maven-systemtest/src/test/resources/testdata/systemtest/Devon4JTemplateTest/TestEntityInputDataaccessGeneration/pom.xml:40]: So should every test have a dependency for an old version and the new version or only the old version?

**Pull request merged**